rules/prefer-global-this.js

```diff
@@ -0,0 +1,259 @@
+	'postMessage',
+
+	// Events commonly associated with "window"
+	...[...windowSpecificEvents].map(event => `on${event}`),
+
+	// To add/remove/dispatch events that are commonly associated with "window"
+	// https://www.w3.org/TR/DOM-Level-2-Events/events.html#Events-flow
+	'addEventListener',
+	'removeEventListener',
+	'dispatchEvent',
+
+	// https://dom.spec.whatwg.org/#idl-index
+	'event', // Deprecated and quirky, best left untouched
+
+	// https://drafts.csswg.org/cssom-view/#idl-index
+	'screen',
+	'visualViewport',
+	'moveTo',
+	'moveBy',
+	'resizeTo',
+	'resizeBy',
+	'innerWidth',
+	'innerHeight',
+	'scrollX',
+	'pageXOffset',
+	'scrollY',
+	'pageYOffset',
+	'scroll',
+	'scrollTo',
+	'scrollBy',
+	'screenX',
+	'screenLeft',
+	'screenY',
+	'screenTop',
+	'screenWidth',
+	'screenHeight',
+	'devicePixelRatio',
+]);
+
+const webWorkerSpecificAPIs = new Set([
+	// https://html.spec.whatwg.org/multipage/workers.html#the-workerglobalscope-common-interface
+	'addEventListener',
+	'removeEventListener',
+	'dispatchEvent',
+
+	'self',
+	'location',
+	'navigator',
+	'onerror',
+	'onlanguagechange',
+	'onoffline',
+	'ononline',
+	'onrejectionhandled',
+	'onunhandledrejection',
+
+	// https://html.spec.whatwg.org/multipage/workers.html#dedicated-workers-and-the-dedicatedworkerglobalscope-interface
+	'name',
+	'postMessage',
+	'onconnect',
+]);
+
+/**
+Report the node with a message.
+
+@param {import('eslint').Rule.RuleContext} context
+@param {import('estree').Node} node
+@param {string} value
+*/
+function report(context, node, value) {
+	const fix = (/** @type {import('eslint').Rule.RuleFixer} fixer */ fixer) => fixer.replaceText(node, 'globalThis');
+	context.report({
+		node,
+		messageId: MESSAGE_ID_ERROR,
+		data: {value, replacement: 'globalThis'},
+		fix,
+		suggest: [{messageId: MESSAGE_ID_SUGGESTION, data: {value, replacement: 'globalThis'}, fix}],
+	});
+}
+
+/**
+Handle nodes and check if they should be reported.
+
+@param {import('eslint').Rule.RuleContext} context
+@param {import('estree').Node | Array<import('estree').Node>} nodes
+*/
+function handleNodes(context, nodes) {
+	if (!Array.isArray(nodes)) {
+		nodes = [nodes];
+	}
+
+	for (const node of nodes) {
+		if (node && node.type === 'Identifier' && globalIdentifier.has(node.name)) {
+			const variable = findVariableInScope(context.sourceCode.getScope(node), node.name);
+			if (!variable) {
+				report(context, node, node.name);
+			}
+		}
+	}
+}
+
+/**
+Check if the node is a window-specific API.
+
+@param {import('estree').MemberExpression} node
+@returns {boolean}
+*/
+const isWindowSpecificAPI = node => {
+	if (node.type !== 'MemberExpression') {
+		return false;
+	}
+
+	if (node.object.name !== 'window' || node.property.type !== 'Identifier') {
+		return false;
+	}
+
+	if (windowSpecificAPIs.has(node.property.name)) {
+		if (['addEventListener', 'removeEventListener', 'dispatchEvent'].includes(node.property.name) && node.parent && node.parent.type === 'CallExpression') {
```

Comment:
`node.parent &&` also unnecessary.

We need test `node === node.parent.callee`.

Reply:
Resolved. 7fe5672e672732fe13ad835791f547c3352bd003